test/benchmark/reasoner/iam/common/Benchmark.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package com.vaticle.typedb.core.reasoner.benchmark.iam.common;
+
+import com.vaticle.typedb.core.common.perfcounter.PerfCounters;
+import com.vaticle.typedb.core.reasoner.common.ReasonerPerfCounters;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.vaticle.typedb.core.common.iterator.Iterators.iterate;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class Benchmark {
+
+    private static final int DEFAULT_NRUNS = 5;
+    private static final double COUNTER_LOWER_MARGIN = 0.75;
+    private static final double COUNTER_UPPER_MARGIN = 1.25;
+    private static final double RUNNING_TIME_UPPER_MARGIN = 3.0;
```

Comment:
Adjusted these values. The running time in particular has a large margin because the numbers were adjusted for my local performance and CI seems to be unto 3x slower.